src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -2994,6 +2994,23 @@ ss::future<> ntp_archiver::apply_spillover() {
     const auto manifest_upload_timeout = _conf->manifest_upload_timeout();
     const auto manifest_upload_backoff = _conf->cloud_storage_initial_backoff();
 
+    // Check the spillover invariant.
+    // The start_offset of the manifest must be equal to the begin_offset of
+    // the first segment in the manifest.
+    if (auto so = manifest().get_start_offset();
+        so.has_value() && !manifest().empty()) {
+        auto fo = manifest().begin()->base_offset;
+        if (fo != so.value()) {
+            vlog(
+              _rtclog.error,
```

Comment:
probably want this at WARN 😄 

https://ci-artifacts.dev.vectorized.cloud/redpanda/73642/0199ba8a-ba60-4d71-9a40-7bbd52eeea28/vbuild/ducktape/results/final/report.html